## Introduction to Python for Machine Learning Quiz Question 11-20
### By Adedamola Babayato
### ID: 157372abee81f000

In [1]:
# Import libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
# Read in the csv file
df = pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv', encoding= 'latin-1')

In [3]:
# check the head info
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [4]:
# check the shape of the dataset
df.shape

(60943, 12)

## Questions

### Question 11:
### What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [9]:
df.groupby('Item')['Y2014','Y2017'].sum()

Y2014      Y2017
Item                                     
Alcohol, Non-Food      2403.00    2348.00
Alcoholic Beverages  102410.11   95581.06
Animal Products       11935.65   11547.65
Animal fats          209460.54  269617.53
Apples and products    9499.23   10198.90
...                        ...        ...
Vegetables, Other    155038.96  157752.59
Vegetal Products     107145.19  107655.20
Wheat and products   232670.13  240047.62
Wine                   4497.36    4178.02
Yams                 200396.96  229174.59

[119 rows x 2 columns]

The total sum of Animal Fat produced in 2014 and 2017 is 209460.54 and 269617.53 respectively.

### Question 12:
### What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [29]:
print(round(df['Y2015'].mean(), 3))
print(round(df['Y2015'].std(), 3))

135.236
1603.404


The mean for the year 2015 is 135.236 while the standard deviation is 1603.404.

### Question 13:
### What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [11]:
print(df['Y2016'].isna().sum())
print(round(df['Y2016'].isna().sum() * 100 / len(df), 2))

1535
2.52


The total number of missing data in 2016 is 1535 while the percentage of the missing data is 2.52%.

### Question 14:
### Which year had the highest correlation with ‘Element Code’? 

In [12]:
df.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

2014 had the highest correlation with Element Code (0.024457)

### Question 15: What year has the highest sum of Import Quantity?
### Question 16: What is the total number of the sum of Production in 2014?

In [13]:
df.groupby('Element')['Y2014','Y2015','Y2016', 'Y2017', 'Y2018'].sum()

Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein supply quantity (g/capita/day)    11842.45    11833.56  
Residuals                                 35500.00    34864.00  
Seed                                      24870.14    25263.14  
Stock Variation                           54316.91    20577.91  
Total Population - Both sexes           1112641.00  1140605.00  
Tourist consumption                          91.00       90.00

2017 had the highest sum of Import quantity (294559.09), while the total sum of production for the year 2014 was 1931287.75.

### Question 17:
### Which of these elements had the highest sum in 2018?

In [14]:
df[['Y2018', 'Element']].groupby('Element').sum().sort_values(by='Y2018',ascending=False).head()

Y2018
Element                                  
Domestic supply quantity       2161192.10
Production                     2075072.89
Food                           1303841.28
Total Population - Both sexes  1140605.00
Food supply (kcal/capita/day)   455261.00

In 2018 Domestic supply quantity had the higest sum of 2161192.10

### Question 18:
### Which of these elements had the 3rd lowest sum in 2018?

In [15]:
df[['Y2018', 'Element']].groupby('Element').sum().sort_values(by='Y2018').head()

Y2018
Element                                         
Tourist consumption                        90.00
Fat supply quantity (g/capita/day)      10258.69
Protein supply quantity (g/capita/day)  11833.56
Stock Variation                         20577.91
Seed                                    25263.14

In 2018 Protein supply quantity (g/capita/day) had the 3rd lowest sum of 11833.56

### Question 19:
### What is the total Import Quantity in Algeria in 2018?

In [38]:
df[df['Area']=='Algeria'].groupby('Element')['Y2018','Area'].sum()

Y2018
Element                                         
Domestic supply quantity                90023.42
Export Quantity                           811.54
Fat supply quantity (g/capita/day)        309.09
Feed                                    10668.00
Food                                    63736.33
Food supply (kcal/capita/day)           13288.00
Food supply quantity (kg/capita/yr)      1509.28
Import Quantity                         36238.29
Losses                                   7708.00
Other uses (non-food)                    3970.08
Processing                               2479.00
Production                              60766.67
Protein supply quantity (g/capita/day)    367.28
Residuals                                 630.00
Seed                                      825.00
Stock Variation                          6172.00
Total Population - Both sexes           42228.00

In 2018 Algeria had a total Import Quantity of 36238.29 

### Question 20:
### What is the total number of unique countries in the dataset?

In [27]:
df.Area.nunique()

49

There are 49 unique countries in the dataset